### PR TITLE
testing/keepassxc: new aport (edited/fixed)

### PR DESCRIPTION
**Pull request by orsonteodoro**
I couldn't revive the deleted fork so this is recreated.  The old comments are at #3134

### CODE REVIEW

File: testing/keepassxc/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Contributor: Orson Teodoro <orsonteodoro@hotmail.com>
+# Maintainer: Orson Teodoro <orsonteodoro@hotmail.com>
+pkgname=keepassxc
+pkgver=2.2.4
+pkgrel=0
+pkgdesc="KeePassXC is a cross-platform community-driven port of the Windows application Keepass Password Safe."
+url="https://keepassxc.org/"
+arch="all"
+license="GPL-2.0 GPL-3.0 BSD-3-Clause CC0-1.0 LGPL-2.1 LGPL-3.0 Nokia-Qt-exception-1.1 MIT"
+_license_build="$license BSL-1.0"
```
Comment (jirutka) [inline testing/keepassxc/APKBUILD]: This variable is not used anywhere, why is it here?

File: testing/keepassxc/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Contributor: August Klein <amatcoder@gmail.com>
+# Contributor: Orson Teodoro <orsonteodoro@hotmail.com>
+# Maintainer: Orson Teodoro <orsonteodoro@hotmail.com>
+pkgname=keepassxc
+pkgver=2.2.4
+pkgrel=0
+pkgdesc="KeePassXC is a cross-platform community-driven port of the Windows application Keepass Password Safe."
+url="https://keepassxc.org/"
+arch="all"
+license="GPL-2.0 GPL-3.0 BSD-3-Clause CC0-1.0 LGPL-2.1 LGPL-3.0 Nokia-Qt-exception-1.1 MIT"
```
Comment (jirutka) [inline testing/keepassxc/APKBUILD]: Please specify exact variant of GPL licenses – `GPL-x.y-only`, or `GPL-x.y-or-later`.

**Comment by jirutka** [comment]
Please add fixed to this PR (branch), **do not close it and create new PR**.

**Comment by sroracle** [comment]
I made a [similar aport for Adélie](https://code.foxkit.us/sroracle/packages/commit/00c1f17da34cfd220d56355f26490e4c7959daa8) and it passed its test suite there, so maybe add a `check()` for `make test`.

File: testing/keepassxc/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# Contributor: Orson Teodoro <orsonteodoro@hotmail.com>
+# Contributor: Max Rees <maxcrees@me.com>
+# Maintainer: Orson Teodoro <orsonteodoro@hotmail.com>
+pkgname=keepassxc
+pkgver=2.2.4
+pkgrel=0
+pkgdesc="KeePassXC is a cross-platform community-driven port of the Windows application Keepass Password Safe."
+url="https://keepassxc.org/"
+arch="all"
+license="GPL-2.0 GPL-3.0 BSD-3-Clause CC0-1.0 LGPL-2.1-only LGPL-3.0-only Nokia-Qt-exception-1.1 MIT BSL-1.0"
```
Comment (kaniini) [inline testing/keepassxc/APKBUILD]: Please use `GPL-2.0+` or `GPL-2.0-or-later` (your preference) for both the GPL2 and GPL3 ones, unless they are GPL-2.0-only and GPL-3.0-only, in which case we cannot accept the package at all as the GPL2 and GPL3 are incompatible with each other when non-upgradeable.

**Comment by kaniini** [comment]
licenses is still wrong

File: testing/keepassxc/APKBUILD (hunk 3 above)
Comment (orsonteodoro) [reply testing/keepassxc/APKBUILD]: The LICENSE.txt (https://github.com/keepassxreboot/keepassxc/blob/148dbc5d41b8f7828c6ae730a92f5d97c13454a7/COPYING) says the art assets keepassxc.png is LGPL-2 (implying only), share/icons/database/* LGPL-2.1 (implying only), share/icons/application/* is LGPL-3.0+.  The code assets are src/streams/qtiocompressor* (case insensitive (LGPL-2 [imply only] or GPL-3; src/gui/KMessageWidget (LGPL 2.1 [implying only with SPDX v2]).

Comment (orsonteodoro) [reply testing/keepassxc/APKBUILD]: The main project is GPL-2 or GPL-3 without plus.

Comment (kaniini) [reply testing/keepassxc/APKBUILD]: In this case, we will take the manual and main.cpp under `GPL-3.0-only`, so the `GPL-2.0` can be removed.

We can take qtiocompressor under `GPL-3.0-or-later`.  The `LGPL-3.0` should become `LGPL-3.0+` or `LGPL-3.0-or-later`.

Comment (kaniini) [reply testing/keepassxc/APKBUILD]: Please update $license accordingly, and I will merge this with the differences in the adelie package and commit it for you.

**Comment by sroracle** [comment]
Can probably be bumped to 2.3.1 if we're still merging this, but needs to add `makedepends="argon2-dev"` and `subpackages="$pkgname-doc"`

**Pull request closed without merging**